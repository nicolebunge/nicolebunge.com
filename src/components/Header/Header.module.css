.header {
  align-items: center;
  background-color: var(--color-background);
  background-image: linear-gradient(to right, #f4f4f4 1px, transparent 1px);
  background-position: 50%;
  background-size: calc(90% / 3);
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  left: 0;
  padding: var(--space-xs) 5%;
  position: fixed;
  right: 0;
  top: 0;
  will-change: transform;
  z-index: 1;
}

@media (min-width: 640px) {
  .header {
    padding-bottom: 0;
    padding-top: 0;
  }
}

@media (min-width: 1440px) {
  .header {
    background-size: calc(var(--grid-max-width) / 3);
    padding-left: 3.5em;
    padding-right: 3.5em;
  }
}
