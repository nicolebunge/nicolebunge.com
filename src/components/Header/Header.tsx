import clsx from 'clsx';
import React, { HTMLAttributes, ReactNode } from 'react';
import { header } from './Header.module.css';

interface HeaderProps extends HTMLAttributes<HTMLDivElement> {
  children: ReactNode;
}

function Header(props: HeaderProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <header className={clsx(className, header)} role="banner" {...otherProps} />;
}

export type { HeaderProps };
export default Header;
