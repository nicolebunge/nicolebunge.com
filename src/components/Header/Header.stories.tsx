import Header from '.';

const meta = {
  component: Header,
};

const Default = {
  args: {
    children: 'Header',
  },
};

export { Default };
export default meta;
