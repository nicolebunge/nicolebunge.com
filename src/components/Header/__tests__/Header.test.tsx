import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import React from 'react';
import * as stories from '../Header.stories';

const { Default } = composeStories(stories);

describe('Header component', () => {
  it('renders correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
