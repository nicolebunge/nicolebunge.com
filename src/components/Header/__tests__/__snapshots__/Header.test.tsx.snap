// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders correctly 1`] = `
<header
  class="header"
  role="banner"
>
  Header
</header>
`;
